pahfit/base.py

```diff
@@ -265,9 +313,27 @@ def plot(self, ax, x, y, model):
         # get the extinction model (probably a better way to do this)
         for cmodel in model:
             if isinstance(cmodel, S07_attenuation):
-                ax_att.plot(x, cmodel(x), "k--")
-                ext_model = cmodel(x)
+                ext_components_S07 = cmodel
 
+        # get additional extinction components that can be
+        # characterized by functional forms (Drude profile in this case)
+        ext_components_funct = []
+        for cmodel in model:
+            if isinstance(cmodel, att_Drude1D):
+                ext_components_funct.append(cmodel)
+
+        if len(ext_components_funct) == 0:
+            ext_model = ext_components_S07(x)
+        elif len(ext_components_funct) == 1:
+            ext_funct_model = ext_components_funct[0]
+            ext_model = ext_components_S07(x) * ext_funct_model(x)
+        elif len(ext_components_funct) >= 2:
+            ext_funct_model = ext_components_funct[0]
+            for cmodel in ext_components_funct[1:]:
+                ext_funct_model *= cmodel
+            ext_model = ext_components_S07(x) * ext_funct_model(x)
```

Comment:
This should be constructed in a loop to allow for an arbitrary number of extinction features.  See how the continuum model is created.

Reply:
```
ext_model = ext_components_S07(x)
for cmodel in ext_components_funct:
     ext_model *= cmodel
```

Reply:
Thanks! Code has been reduced to:         
```
ext_model = ext_components_S07(x)
if len(ext_components_funct) != 0:
    for cmodel in ext_components_funct:
        ext_model *= cmodel(x)
```